hippo-digital/svp-tech-poc
# Task/add persistence to dynamodb

## Summary of changes

The submit button now works, and results in the contents of the form being persisted to a dynamo-db table named `svp-development`. The NHS login `sub` is included in the recorded data, and has a secondary index associated with its entry in the table.

## Testing instructions

Run the following commands:

```bash
docker pull amazon/dynamodb-local
docker run -d -p 8000:8000 amazon/dynamodb-local
```

Then fire up the app as usual, log in, and fill out the presented form. After hitting 'Submit' on the check your answers page the record will be pushed to the database and you will be routed back to the index page.

The simplest way to view the contents of the table (to ensure that the result is in fact persisted) is with the command:

```bash
aws dynamodb scan --table-name svp-development --endpoint-url http://localhost:8000 --region 'eu-west-2'
```

You may need to install `aws-cli` before running the above command.
